Makefile

```diff
@@ -31,25 +32,34 @@ all: docker
 gobuild:
 	./scripts/build false
 
+# create output directories
+out:
+	mkdir ./out
+	mkdir ./out/test-artifacts
+	mkdir ./out/cni-plugins
+
 # Basic go build
 static:
 	./scripts/build
 
+builder-image:
+	@docker build -f scripts/dockerfiles/Dockerfile.build -t "amazon/amazon-ecs-agent-build:make" .
+
 # 'build-in-docker' builds the agent within a dockerfile and saves it to the ./out
 # directory
-build-in-docker:
-	@docker build -f scripts/dockerfiles/Dockerfile.build -t "amazon/amazon-ecs-agent-build:make" .
+build-in-docker: builder-image out
 	@docker run --net=none \
-	  -e TARGET_OS="${TARGET_OS}" \
-	  -e LDFLAGS="-X github.com/aws/amazon-ecs-agent/agent/config.DefaultPauseContainerTag=$(PAUSE_CONTAINER_TAG) \
-	  -X github.com/aws/amazon-ecs-agent/agent/config.DefaultPauseContainerImageName=$(PAUSE_CONTAINER_IMAGE)" \
-	  -v "$(PWD)/out:/out" \
-	  -v "$(PWD):/go/src/github.com/aws/amazon-ecs-agent" \
-	  "amazon/amazon-ecs-agent-build:make"
+		-e TARGET_OS="${TARGET_OS}" \
```

Comment:
Would you be willing to change to the long flags for these?  `--env`, `--volume`, etc